src/did/cache.ts

```diff
@@ -0,0 +1,43 @@
+import NodeCache from 'node-cache';
+
+/**
+ * A generalized cache interface.
+ */
+export interface Cache {
+  /**
+   * Sets a key-value pair.
+   * @returns `true` if key-value pair is stored successfully; `false` otherwise
+   */
+  set(key: string, value: any): Promise<boolean>;
+
+  /**
+   * Gets the value corresponding to the given key.
+   * @returns value stored corresponding to the given key; `undefined` if key is not found or expired
+   */
+  get(key: string): Promise<any | undefined>;
+}
+
+/**
+ * A cache using local memory.
+ */
+export class MemoryCache implements Cache {
```

Comment:
do you think it makes sense to put this into `src/utils` instead? it's so general purpose that we could definitely use it whenever an in-mem cache is needed

Reply:
Sure, will move over to `utils`.